scripts/sonar-scan.sh

```diff
@@ -22,6 +22,12 @@
 # docker-compose ones are:
 #  - MRAA_SRC_DIR - path to mraa's git clone in the Docker container.
 
+# Check required environment variables and exit if they are not set
+MRAA_SRC_DIR=${MRAA_SRC_DIR:?value not provided}
+SONAR_PROJ_KEY=${SONAR_PROJ_KEY:?value not provided}
+SONAR_ORG=${SONAR_ORG:?value not provided}
+SONAR_TOKEN=${SONAR_TOKEN:?value not provided}
+
```

Comment:
+1 LGTM. One nitpick-level suggestion would be to add a variable name to the message to make it easier to detect what's missing, but that's really optional.